examples/platform/silabs/SiWx917/SiWxWifiInterface.cpp

```diff
@@ -69,89 +77,368 @@ extern "C" {
+        wfx_rsi.ap_chan = scan_result->scan_info[0].rf_channel;
+        memcpy(&wfx_rsi.ap_mac.octet, scan_result->scan_info[0].bssid, BSSID_LEN);
     }
 
-    status = osTimerStop(sDHCPTimer);
-    if (status != osOK)
+    osSemaphoreRelease(sScanCompleteSemaphore);
+    return status;
+}
+#endif
+
+sl_status_t InitiateScan()
+{
+    sl_status_t status = SL_STATUS_OK;
+
+// TODO: this changes will be reverted back after the Silabs WiFi SDK team fix the scan API
+#ifndef EXP_BOARD
+    sl_wifi_ssid_t ssid = { 0 };
+
+    // TODO: this changes will be reverted back after the Silabs WiFi SDK team fix the scan API
+    sl_wifi_scan_configuration_t wifi_scan_configuration = default_wifi_scan_configuration;
+
+    ssid.length = wfx_rsi.sec.ssid_length;
+
+    // TODO: workaround because the string management with the null termination is flawed
+    chip::Platform::CopyString((char *) &ssid.value[0], ssid.length + 1, wfx_rsi.sec.ssid);
+    sl_wifi_set_scan_callback(ScanCallback, NULL);
+
+    osSemaphoreAcquire(sScanInProgressSemaphore, osWaitForever);
+
+    // This is an odd success code?
+    status = sl_wifi_start_scan(SL_WIFI_CLIENT_2_4GHZ_INTERFACE, &ssid, &wifi_scan_configuration);
+    if (status == SL_STATUS_IN_PROGRESS)
     {
-        ChipLogError(DeviceLayer, "CancelDHCPTimer: failed to stop timer: %d", status);
+        osSemaphoreAcquire(sScanCompleteSemaphore, kWifiScanTimeoutTicks);
+        status = SL_STATUS_OK;
     }
+    else
+    {
+        osSemaphoreRelease(sScanInProgressSemaphore);
+    }
+#endif
+
+    return status;
 }
 
-static void StartDHCPTimer(uint32_t timeout)
+sl_status_t SetWifiConfigurations()
 {
-    osStatus_t status;
+    sl_status_t status = SL_STATUS_OK;
 
-    // Cancel timer if already started
-    CancelDHCPTimer();
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+    // Setting the listen interval to 0 which will set it to DTIM interval
+    sl_wifi_listen_interval_t sleep_interval = { .listen_interval = 0 };
+    status                                   = sl_wifi_set_listen_interval(SL_WIFI_CLIENT_INTERFACE, sleep_interval);
+    VerifyOrReturnError(status == SL_STATUS_OK, status);
+
+    sl_wifi_advanced_client_configuration_t client_config = { .max_retry_attempts = 5 };
+    status = sl_wifi_set_advanced_client_configuration(SL_WIFI_CLIENT_INTERFACE, &client_config);
+    VerifyOrReturnError(status == SL_STATUS_OK, status);
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+
+    status = sl_net_set_credential(SL_NET_DEFAULT_WIFI_CLIENT_CREDENTIAL_ID, SL_NET_WIFI_PSK, &wfx_rsi.sec.passkey[0],
+                                   wfx_rsi.sec.passkey_length);
+    VerifyOrReturnError(status == SL_STATUS_OK, status);
+
+    sl_net_wifi_client_profile_t profile = {
+        .config = {
+            .ssid = {
+                //static cast because the types dont match
+                .length = static_cast<uint8_t>(wfx_rsi.sec.ssid_length),
```

Comment:
What type is wfx_rsi.sec.ssid_length? Will probably never get a 255+ char ssid. But there is a risk out of bound write with the memcpy at line 396

Reply:
Underlying wifi api don't support an SSID longuer than 255 under the hood. Any larger wouldn't work in this case.
This are sompe the pieces that are going to get deleted because the typing doesn't work between our integration and the wifi sdk 

```
typedef struct {
  uint8_t value[32]; ///< SSID value
  uint8_t length;    ///< Length of the SSID
} sl_wifi_ssid_t;
```